src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown

```diff
@@ -0,0 +1,295 @@
+featured_image: /web/updates/images/2016-03-02-access-usb-devices-on-the-web/web-usb-hero.jpg
+published_on: 2016-03-02
+updated_on: 2016-03-30
+authors:
+  - beaufortfrancois
+tags:
+  - news
+  - USB
+  - WebUSB
+  - IoT
+  - Arduino
+---
+
+If I tell you plain and simple "USB", there are good chances that you will
+immediately think of keyboards, mice, audio, video and storage devices. You're
+right but you'll find other kinds of Universal Serial Bus (USB) devices out
+there.
+
+These non-standardized USB devices require hardware vendors to write native
+drivers and SDKs in order for you (developer) to take advantage of them. Sadly
+this native code has historically prevented these devices from being used by
+the Web. And that's exactly why the WebUSB API has been created: to **provide
+an easy way to safely expose USB device services to the Web**. With this API,
+hardware manufacturers will be able to build cross-platform JavaScript SDKs for
+their devices.
+
+## Before we start
+
+This article assumes you have some basic knowledge of how USB works. If not, I
+recommend reading [USB in a NutShell](http://www.beyondlogic.org/usbnutshell).
+For background information about USB, check out the [official USB
+specifications](http://www.usb.org/home).
+
+The [WebUSB API](https://wicg.github.io/webusb/) draft is hmm... a
+draft. That's why the Chrome Team is actively looking for eager developers to
+give it a try and send
+[feedback on the spec](https://github.com/wicg/webusb/issues) and
+[feedback on the implementation](https://bugs.chromium.org/p/chromium/issues/entry?components=Blink%3EUSB).
+
+At the time of writing, the WebUSB API is partially implemented behind an
+experimental flag in Chrome for Windows, Mac OS, Linux and Chrome OS. Go to
+`chrome://flags/#enable-experimental-web-platform-features`, enable the
+highlighted flag, restart Chrome and you should be good to go.
+
+<img style="width:723px; max-height:205px" src="/web/updates/images/2016-03-02-access-usb-devices-on-the-web/web-usb-flag.png" alt="Web USB Flag highlighted in chrome://flags"/>
+
+## Privacy & Security
+
+### Attacks against USB devices
+
+The WebUSB API does not even try to provide a way for any web page to
+connect to arbitrary USB devices. There are plenty of published attacks against
+USB devices that makes it unsafe to allow this.
+
+For this reason, a USB device can define a set of origins that are allowed to
+connect to it. This is similar to the
+[CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS)
+mechanism. In other words, **WebUSB devices are associated with a web origin**
+and can only be accessed from a page from the same origin.
+
+### HTTPS Only
+
+Because this experimental API is a powerful new feature added to the Web,
+Chrome aims to make it available only to [secure
+contexts](https://w3c.github.io/webappsec/specs/powerfulfeatures/#intro). This
+means you'll need to build with TLS in mind.
```

Comment:
I dislike all this talk about TLS and HTTP only... its actually a "secure origin" or something like that, all this talk of TLS etc puts people off as it makes people think they need TLS locally to develop these things, which just isn't the case